src/main/java/io/github/mzmine/modules/io/import_rawdata_mzml/msdk/data/BuildingMzMLMsScan.java

```diff
@@ -747,24 +578,55 @@ public Optional<Integer> getScanNumber(String spectrumId) {
     return Optional.ofNullable(null);
   }
 
-  public void setAdvancedParameters(ParameterSet advancedParam) {
-    this.advancedParameters = advancedParam;
-  }
+  /**
+   * Called when spectrum end is read. Load, process data points and memory map resulting data to
+   * disk to save RAM.
+   */
+  public boolean loadProcessMemMapData(final MemoryMapStorage storage,
+      final MsProcessorList spectralProcessor) {
+    try {
+      var specData = loadData();
+      if (specData == null) {
+        // may be null for UV spectra
+        return false;
+      }
+
+      // process and filter - needs metadata so wrap
+      specData = spectralProcessor.processScan(this, specData);
+
+      // memory map
+      this.mzValues = StorageUtils.storeValuesToDoubleBuffer(storage, specData.mzs());
```

Comment:
maybe nullify SimpleSpectralArrays after, we're not using it anymore, right? Though GC probably will handle it anyway

Reply:
its method internal so its freed directly. no need to null